src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +164,213 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
+                errorMessageBuilder.Append(detailsMessage);
+            
+            TryAbortTransactionAfterException(errorMessageBuilder, out _);
+            throw new QueueLoadException(errorMessageBuilder.ToString(), e);
         }
         catch (Exception ex)
         {
-            try
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to produce message.{Environment.NewLine}");
+            TryAbortTransactionAfterException(errorMessageBuilder, out _);
+            throw new QueueLoadException(ex.Message, ex);
+        }
+        
+        
+        try
+        {
+            _producer.CommitTransaction();
+        }
+        catch (KafkaTxnRequiresAbortException ex)
+        {
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Commit transaction failed, due to {nameof(KafkaTxnRequiresAbortException)}. ");
+            if (TryToGetAclErrorsDetails(out var detailsMessage))
+                errorMessageBuilder.Append(detailsMessage);
+
+            if (TryAbortTransactionAfterException(errorMessageBuilder, out var exception) == false)
+                throw new AggregateException(errorMessageBuilder.ToString(), ex, exception);
+            
+            errorMessageBuilder.Append("Although transaction is lost, required transaction abort was handled successfully.");
+            throw new QueueLoadException(errorMessageBuilder.ToString(), ex);
+        }
+        catch (KafkaException ex)
+        {
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Commit transaction failed. ");
+            if (TryToGetAclErrorsDetails(out var detailsMessage))
+                errorMessageBuilder.Append(detailsMessage);
+            
+            TryAbortTransactionAfterException(errorMessageBuilder, out _);
+            throw new QueueLoadException(errorMessageBuilder.ToString(), ex);
+        }
+
+        return count;
+    }
+
+    private IAdminClient GetAdminClient()
+    {
+        var config = new AdminClientConfig
+        {
+            BootstrapServers = Configuration.Connection.KafkaConnectionSettings.BootstrapServers,
+        };
+        QueueBrokerConnectionHelper.SetupKafkaClientConfig(config, Configuration.Connection.KafkaConnectionSettings);
+        return new AdminClientBuilder(config).Build();
+    }
+
+
+    private static IEnumerable<AclBinding> GetMatchingAllowAclBindings(IAdminClient adminClient, AclOperation operation = AclOperation.All,
+        ResourceType resourceType = ResourceType.Any)
+    {
+        var aclBindingsForTransactionIdsTask = adminClient.DescribeAclsAsync(new AclBindingFilter
+        {
+            EntryFilter = new AccessControlEntryFilter {Operation = operation, PermissionType = AclPermissionType.Allow},
+            PatternFilter = new ResourcePatternFilter {ResourcePatternType = ResourcePatternType.Any, Type = resourceType}
+        });
+
+        aclBindingsForTransactionIdsTask.Wait();
```

Comment:
Please use `AsyncHelpers.RunSync()` instead to avoid potential deadlock

Reply:
done